### PR TITLE
Introduce `FluxCountMapMathToIntExact` Refaster rule

### PR DESCRIPTION
**Pull request by Ptijohn**
Suggested commit message:
```
Introduce `FluxCount` Refaster rules (#516)
```

### CODE REVIEW

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java [hunk 1]
```diff
@@ -805,6 +809,20 @@ Flux<T> after(Flux<? extends Iterable<T>> flux, int prefetch) {
     }
   }
 
+  /** Prefer {@link Flux#count()} over collecting into a list and counting its elements. */
+  static final class FluxCount<T> {
```
Comment (Stephan202) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java]: While it's an unwieldy name, likely we'll eventually name this rule `FluxCountMapMathToIntExact` (based on the `@AfterTemplate` code, so let's go for that.

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java [hunk 2]
```diff
@@ -805,6 +809,20 @@ Flux<T> after(Flux<? extends Iterable<T>> flux, int prefetch) {
...
+    @BeforeTemplate
+    Mono<Integer> before(Flux<T> flux) {
+      return flux.collect(toImmutableList())
```
Comment (Stephan202) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java]: With [`CollectorMutability`](https://github.com/PicnicSupermarket/error-prone-support/blob/6b4fba62da0e53834c839174ef84da29afd386cb/error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/CollectorMutability.java) we also suggest `toCollection(ArrayList::new)` at times; let's add that case.

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java [hunk 3]
```diff
@@ -805,6 +809,20 @@ Flux<T> after(Flux<? extends Iterable<T>> flux, int prefetch) {
...
+          .map(Refaster.anyOf(Collection::size, ImmutableList::size, List::size));
```
Comment (Stephan202) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java]: There's also `ImmutableCollection::size` :smile:

**Review by Stephan202 — APPROVED** [review]
Tnx @Ptijohn! Added a commit with some suggestions. Suggested commit message:
```
Introduce `FluxCountMapMathToIntExact` Refaster rule (#516)
```

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java [hunk 4]
```diff
@@ -805,6 +812,31 @@ Flux<T> after(Flux<? extends Iterable<T>> flux, int prefetch) {
...
+    @BeforeTemplate
+    Mono<Integer> before(Flux<T> flux) {
+      return Refaster.anyOf(
+          flux.collect(toImmutableList())
+              .map(
+                  Refaster.anyOf(
+                      Collection::size,
+                      List::size,
+                      ImmutableCollection::size,
+                      ImmutableList::size)),
```
Comment (Stephan202) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java:829]: Sorted semi-hierarchically :shrug:

**Review by rickie — APPROVED** [review]
Changes LTGM! Nice PR @Ptijohn :rocket:

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java (hunk 3 above)
Comment (Ptijohn) [reply error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java]: Of course! 
Thanks :)

**Pull request merged**